### PR TITLE
k8s: make all ports explicit (and configurable)

### PR DESCRIPTION
**Pull request by hunchback**
# Setup

```
cd contrib/kubernetes
```

# Reconfig

Change analyzer listening port from 8082 to 18082:

 ```
cat skydive.yaml | sed -e s/8082/18082/ | kubectl apply -f -
```

 Change agent listening port from 8081 to 18081:

 ```
cat skydive.yaml | sed -e s/8081/18081/ | kubectl apply -f -
```

 Change elasticsearch port from 9200 to 19200:

 ```
cat skydive.yaml | sed -e s/9200/19200/ | kubectl apply -f -
```

 Change etcd port from 12379/12380 to 22379/22380:

 ```
cat skydive.yaml | sed -e s/12379/22379/ -e s/12380/22380/ | kubectl apply -f -
```

# Test

```
kubectl get pods                                 # check all RUNNING
kubectl get services                            # check correct ports
```

### CODE REVIEW

**Comment by hunchback** [comment]
skip skydive-compile-tests

**Comment by hunchback** [comment]
run skydive-compile-tests

**Comment by hunchback** [comment]
@lebauce  - please review

File: contrib/kubernetes/skydive.yaml [hunk 1]
```diff
@@ -102,6 +98,11 @@ spec:
           initialDelaySeconds: 20
           periodSeconds: 10
           failureThreshold: 10
+        env:
+        - name: SKYDIVE_ANALYZER_LISTEN
+          value: "0.0.0.0:$(SKYDIVE_ANALYZER_SERVICE_PORT_API)"
```
Comment (lebauce) [inline contrib/kubernetes/skydive.yaml]: Where is `SKYDIVE_ANALYZER_SERVICE_PORT_API` defined ?

File: contrib/kubernetes/skydive.yaml [hunk 2]
```diff
@@ -102,6 +98,11 @@ spec:
...
+        - name: SKYDIVE_ETCD_LISTEN
+          value: "0.0.0.0:$(SKYDIVE_ANALYZER_SERVICE_PORT_ETCD)"
```
Comment (lebauce) [inline contrib/kubernetes/skydive.yaml]: Same for `SKYDIVE_ANALYZER_SERVICE_PORT_ETCD`

File: contrib/kubernetes/skydive.yaml (hunk 1 above)
Comment (hunchback) [reply contrib/kubernetes/skydive.yaml]: https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables

File: contrib/kubernetes/skydive.yaml (hunk 2 above)
Comment (hunchback) [reply contrib/kubernetes/skydive.yaml]: https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables

**Pull request merged**